turbopack/crates/turbopack-core/src/version.rs

```diff
@@ -191,6 +191,7 @@ pub enum Update {
 #[derive(PartialEq, Eq, Debug, Clone, TraceRawVcs, ValueDebugFormat, NonLocalValue)]
 pub struct TotalUpdate {
     /// The version this update will bring the object to.
+    // TODO: This trace_ignore is *very* wrong, and could cause problems if/when we add a GC
```

Comment:
It's very unsound, but currently correct. All values that implement the `Version` trait, must not contain `Vcs` (they must be flat. This is a requirement of the `Version` trait, which represents a frozen version. It can't contain Vcs as they might change over-time, which is not what `Version` represents.

We should probably at least document that on the `Version` trait. Cool would be some kind of `FrozenValue` trait, but it was not worth it for the few cases where we use that (`Version, `PlainIssue`, `PlainDiagnostic`).

Reply:
It's actually not correct because `EcmascriptDevChunkListVersion` contains a `ResolvedVc`, but I suppose the blame for that is on `EcmascriptDevChunkListVersion`, not on this code.

I'll make some updates to the comments and document this on `Version` though!

Enforcing this statically would be rather painful, as it would require adding yet another marker trait for values with no vc (resolved, operation, or otherwise).